Vitor Kawai Sala

vkawai opened the pull request.

vkawai committed:
Bronze 1 e Bronze 2

vkawai committed:
Arrumado o Bronze 1 e 2

vkawai committed:
Bronze 3

vkawai commented:
Commit para o desafio Bronze 3

ViniciusMiana commented on iTunesSearch/Entidades/Ebook.h:
```diff
@@ -0,0 +1,21 @@
+
+@interface Ebook : NSObject
+
+@property (nonatomic, strong) NSString *titulo;
```
O que ebook, musica, podcast, etc,,, tem em comum? Não daria para usar herança?

vkawai replied:
Sim, é perfeitamente possível usar herança. Não tinha pensado nisso

ViniciusMiana commented on iTunesSearch/TableViewController.m:
```diff
@@ -29,9 +32,19 @@ - (void)viewDidLoad {
     
     iTunesManager *itunes = [iTunesManager sharedInstance];
     midias = [itunes buscarMidias:@"Apple"];
```
So busca Apple, para que serve a caixa de texto?

vkawai replied:
Essa primeira parte, só irá retornar uma busca "default" (aplicativo já iniciar com algum resultado já em exibição), a caixa de texto faz busca de outros termos e substitue a busca da palavra chave "apple" ou algum outro termo anterior.

ViniciusMiana commented on iTunesSearch/TableViewController.m:
```diff
@@ -61,8 +77,67 @@ -(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NS
+
+}
+
+-(void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPath *)indexPath{
```
Qual a ideia aqui?

vkawai replied:
A idéia é que, ao clicar em uma célula, ela se expande mostrando todos os dados adicionais.
Quando a célula perde o status de selecionado, ela retornará ao estado original

vkawai committed:
Correções nos exercícios Bronze 1,2 e 3

vkawai committed:
Prata 1 e Prata 2

vkawai commented:
Commit para os desafios prata 1 e prata 2

Nota: iTunes está fora do ar, testes incompletos.

vkawai committed:
Feito todo o resto (Bronze 4, prata 3, ouro 1, ouro 2, ouro 3)

vkawai committed:
Refatoração de código

vkawai committed:
Refatoração (de novo) e UIActivityIndicatorView funcionando

vkawai committed:
Refatoração e implementação.

- Pequena limpeza de código para tirar blocos de execução em situações
  síncronas.
- Implementado sistema de cache de imagem simples.
- Implementado indicador de atividade usando UIView ao invés de
  UIAlertView, permitindo maior customização.

vkawai committed:
tentando de novo....

vkawai committed:
Esqueci de tirar o sleep...

vkawai committed:
Arrumando a internacionalização (traduções) e limpeza de código
